tests/test_run_reference_vmm.py

```diff
@@ -138,6 +139,43 @@ def expect_string(vmm_process, expected_string, timeout=TEST_TIMEOUT):
             raise
 
 
+def run_cmd_inside_vm(cmd, vmm_process, prompt, timeout=5):
```

Comment:
With adding this function we now use multiple ways of sending commands to the vmm. Is it possible to unify them? For example, by using this utility also in the tests that check the vcpu & memory?

Reply:
Actually, when we use `pexpect` a lot of this code is just going to go away (or will change substantially), so it's not a very good investment at the moment to clean that up.

I looked at the `pexpect` code which does something very similar to the above 'hand cooked' non-blocking (in a better and proven way!), so likely that's the path to be taken at some point of time soon enough.

Reply:
Hmmm, in this case I wonder if we need to go through this intermediate stage in which we use `run_cmd_inside_vm`. It can be confusing when reading the tests to see that some use one way of executing commands, and other use a different way.

Reply:
The function would still be called `run_cmd_inside_vm` - except internally it will use `pexpect` instead of the `while True` loop. The function will still have (mostly) identical signature. But to be able to do that first we need to change the `create_vmm_process`, which I think is a big change and quite unrelated to this specific issue! (block device test).

Yes, initially my thoughts were same, these two ways are different and confusing. May be a good idea is to add a `FIXME` to those functions now and leave them as it is right now and fix them along with the implementation of `pexpect`. 

I will  try to work on a  PR using `pexpect` sometime next week.